### ISSUE TITLE
connect() call occasionally panic

### ISSUE THREAD

**Issue by MrSpock** [mwylde/rumble#23]
Ocassionally from time to time when calling Peripheral connect() methon I'm getting panic 
`thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Other("ENOSYS: Function not implemented")', libcore/result.rs:1009:5`

I'm not unwrapping anything:

     match periph.connect() {
         Ok(_) => info!("Connected ! .."),
         Err(e) => {
             info!("Unable to connect !:{}", e);
             process::exit(-1);
         }
     }

It looks like some error from undelying libc* bindings ..?

**Comment by MrSpock** [comment]
Hi,
Happy new year !!.
Please find attached two panics I was able to catch..
[rumble-panic_02.txt](https://github.com/mwylde/rumble/files/2719026/rumble-panic_02.txt)
[rumble-panic_01.txt](https://github.com/mwylde/rumble/files/2719027/rumble-panic_01.txt)

**Comment by mwylde** [comment]
Hey @MrSpock, thanks for the report. This particular panic should have been fixed in 52a18bd551b50da1d735f41f5ed7a1ea2117db3b, which correctly returns the err instead of unwrapping it. Can you try updating to master and see if that fixes it?

**Comment by MrSpock** [comment]
Now I can handle this panic but it seems that BLE Linux stuck somehow sucks.
Whtn this error occurs and when I'm handling it in a loop and trying another connect() ends up with:
```
Unable to connect !:ENOSYS: Function not implemented
Unable to connect !:ENOMEM: Out of memory
```
And then everything hangs ..
But this looks like underlying BLE stack issue
